File: config/src/artifact/vorpal_user.rs
use crate::artifact::SYSTEMS;
use anyhow::Result;
use vorpal_sdk::{artifact::UserEnvironment, context::ConfigContext};

#[derive(Default)]
pub struct VorpalUser {}

impl VorpalUser {
    pub fn new() -> Self {
        Self::default()
    }

    pub async fn build(self, context: &mut ConfigContext) -> Result<String> {
        UserEnvironment::new("vorpal-user", SYSTEMS.to_vec())
            .with_artifacts(vec![])
            .with_environments(vec!["PATH=$HOME/.vorpal/bin".to_string()])
            .with_symlinks(vec![
                (
                    "$HOME/Development/repository/github.com/ALT-F4-LLC/vorpal.git/main/target/debug/vorpal",
                    "$HOME/.vorpal/bin/vorpal",
                ),
            ])
            .build(context)
            .await
    }
}

File: sdk/go/cmd/vorpal/artifact/vorpal_user.go
package artifact

import (
	"github.com/ALT-F4-LLC/vorpal/sdk/go/pkg/artifact"
	"github.com/ALT-F4-LLC/vorpal/sdk/go/pkg/config"
)

func BuildVorpalUser(context *config.ConfigContext) (*string, error) {
	return artifact.
		NewUserEnvironment("vorpal-user", SYSTEMS).
		WithArtifacts([]*string{}).
		WithEnvironments([]string{"PATH=$HOME/.vorpal/bin"}).
		WithSymlinks(map[string]string{
			"$HOME/Development/repository/github.com/ALT-F4-LLC/vorpal.git/main/target/debug/vorpal": "$HOME/.vorpal/bin/vorpal",
		}).
		Build(context)
}
